protocol-units/bridge/service/src/lib.rs

```diff
@@ -20,6 +20,28 @@ pub mod rest;
 mod states;
 pub mod types;
 
+#[derive(Debug)]
+struct HeathCheckStatus {
+	chain_one: bool,
+	chain_two: bool,
+}
+
+impl HeathCheckStatus {
+	fn new() -> Self {
+		HeathCheckStatus { chain_one: true, chain_two: true }
+	}
+	fn check(&self) -> bool {
```

Comment:
If this is just a logical AND why don't we just have the one bool and and the setter `update_state` simply && the current state instead?

Reply:
There's 2 monitoring loop. I think it's important to be able to detect which loop failed. In the health check return response, when it's ok it return OK, but when it failed, it returns the details. This check is used to know if there's any failed loop or not.